core/global_config.cpp

```diff
@@ -236,13 +241,43 @@ bool GlobalConfig::_load_resource_pack(const String &p_pack) {
 	return true;
 }
 
+static String _find_project_file(DirAccess* p_da, bool p_res = false) {
+	p_da->list_dir_begin();
+	String ret = "";
+	while (true) {
+		bool isdir;
+		String f = p_da->get_next(&isdir);
+		if (f == "")
+			break;
+
+		if (!isdir) {
+			if (f.get_extension() == "godot") {
+
+				if (p_res) {
+					ret = "res://" + f;
+				} else {
+					ret = p_da->get_current_dir() + "/" + f;
+				}
+			}
+		}
+	}
+	p_da->list_dir_end();
+	return ret;
+}
+
+static String _find_project_file() {
+	DirAccess *dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	String ret = _find_project_file(dir, true);
+	memdelete(dir);
+	return ret;
+}
+
 Error GlobalConfig::setup(const String &p_path, const String &p_main_pack) {
 
 	//If looking for files in network, just use network!
-
 	if (FileAccessNetworkClient::get_singleton()) {
-
-		if (_load_settings("res://godot.cfg") == OK || _load_settings_binary("res://godot.cfb") == OK) {
+		String gdproj = _find_project_file();
+		if (_load_settings(gdproj) == OK || _load_settings_binary("res://godot.cfb") == OK) {
```

Comment:
Well, good question. I don't know what kind of performance difference a binary config file makes, I guess @reduz should clear this up :P 
The reason I kept it at `godot.cfb` was that usually neither the game developer nor the player will ever care about this file, so the name doesn't really matter IMO.